.github/workflows/go.yml

```diff
@@ -0,0 +1,84 @@
+name: Go
+
+on:
+  push:
+    tags:
+      - "*"
+  pull_request:
+    branches: ["v2"]
+
+jobs:
+  configure:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+
+  upload:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - name: linux-x86_x64
+            os: linux
+            arch: amd64
+          - name: linux-arm64
+            os: linux
+            arch: arm64
+          - name: linux-arm32
+            os: linux
+            arch: arm
+          - name: windows-amd64
+            os: windows
+            arch: amd64
+          - name: macos-amd64
+            os: darwin
+            arch: amd64
+          - name: macos-arm64
+            os: darwin
+            arch: arm64
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: create build dirs
+        run: mkdir ${{ matrix.name }}
+
+      - name: Build
+        run: GOOS=${{ matrix.os }} GOARCH=${{ matrix.arch }} go build -o ${{ matrix.name }} ./...
+
+      - name: Get Git Version String
+        working-directory: ${{github.workspace}}
+        run: |
+          if ${{ startsWith(github.ref, 'refs/tags/') }}; then
+            echo "app_version=${{ github.ref_name }}" >> $GITHUB_ENV
+          elif [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo "app_version=$(echo "${{ github.head_ref }}" | sed -r 's/\//_/g')-$(git rev-parse --short "${{ github.event.pull_request.head.sha }}")" >> $GITHUB_ENV
+          else
+            echo "app_version=$(git rev-parse --abbrev-ref HEAD | sed -r 's/\//_/g')-$(git describe --always)" >> $GITHUB_ENV
+          fi
+      - name: Print Git Version String
+        run: |
+          echo ${{ env.app_version }}
+      - name: Make build results executable
+        run: chmod +x mqtt-re*
+
+      - name: Zipping
+        run: zip -r mqtt-record-replay-${{ matrix.name }}-${{ env.app_version }}.zip ${{ matrix.name }}
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: mqtt-record-replay-${{ matrix.name }}-${{ env.app_version }}
+          path: mqtt-record-replay-${{ matrix.name }}-${{ env.app_version }}.zip
```

Comment:
upload-artifact already does a zip file, so please just specify the two file paths here.

Reply:
Okay. Good to know. I thought it would just upload files

Reply:
It's a bit stupid though as it does a double-zip now.

Reply:
yes. im changing it